### PR TITLE
[tune] Unflattened lookup for ProgressReporter

### LINKED ISSUE #9307: [Tune] status tables during training aren't showing values for tunable, nested config parameters
### What is the problem?

Ray 0.8.6. Python 3.7.7.

Running this script (actually run with ipython):

```python
from ray import tune
ray.init()
tune.run(
    "PPO",
    stop={"episode_reward_mean": 400},
    config={
        "env": "CartPole-v1",
        "num_gpus": 0,
        "num_workers": 1,
        # "lr": tune.grid_search([0.01, 0.001, 0.0001]),
        "model": {
            'fcnet_hiddens': [tune.grid_search([20, 40, 60, 80]), tune.grid_search([20, 40, 60, 80])]
        },
        "eager": False,
    },
)
```

It appears that Tune easily drives the 16 different combinations for the two elements of `fcnet_hidden` (number of weights for two NN layers). All combinations easily reach the reward of 400, but some do so with lower iterations and/or elapsed times, which is what I want to determine.

The bug is in the status table printed. It has two columns for the tuning parameters, `model/fcnet_hiddens/0` and `model/fcnet_hiddens/1`, but they are empty. I can determine the actual values by looking at the `experiment_tag`, e.g., ` experiment_tag: 8_fcnet_hiddens_0=20,fcnet_hiddens_1=60`

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

`ProgressReporter` calls `_get_trial_info()` to get the data to display. For the lookup, `_get_trial_info()` flattened the `result` and `config` dicts - however, `flatten_dict()` does not flatten lists. This lead to empty lookups for nested config values, as described in #9307.

With this fix, `_get_trial_info()` uses a new util function `unflattened_lookup()` instead, which works the other way around: The lookup keys are split and iteratively queried from the data dict.

## Related issue number

Should fix #9307

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/utils/util.py [hunk 1]
```diff
@@ -216,6 +216,23 @@ def flatten_dict(dt, delimiter="/"):
     return dt
 
 
+def unflattened_lookup(flat_key, lookup, delimiter="/", default=None):
```
Comment (amogkam) [inline python/ray/tune/utils/util.py:219]: Can you add a docstring with an example here.

**Review by amogkam — APPROVED** [review]
Looks good- just left 1 comment

File: python/ray/tune/tests/test_progress_reporter.py [hunk 2]
```diff
@@ -246,21 +246,24 @@ def testProgressStr(self):
             t.trial_id = "%05d" % i
             t.local_dir = "/foo"
             t.location = "here"
-            t.config = {"a": i, "b": i * 2}
-            t.evaluated_params = t.config
+            t.config = {"a": i, "b": i * 2, "n": {"k": [i, 2 * i]}}
+            t.evaluated_params = ["a", "b", "n/k/0", "n/k/1"]
```
Comment (richardliaw) [inline python/ray/tune/tests/test_progress_reporter.py:250]: Are evaluated params usually a list?

File: python/ray/tune/tests/test_progress_reporter.py [hunk 3]
```diff
@@ -246,21 +246,24 @@ def testProgressStr(self):
...
             t.last_result = {
                 "config": {
                     "a": i,
-                    "b": i * 2
+                    "b": i * 2,
+                    "n": {
+                        "k": [i, 2 * i]
+                    }
```
Comment (richardliaw) [inline python/ray/tune/tests/test_progress_reporter.py:257]: I didn't even realize we supported lists.

File: python/ray/tune/tests/test_progress_reporter.py (hunk 2 above)
Comment (krfricke) [reply python/ray/tune/tests/test_progress_reporter.py]: No. For the purpose of the Reporter they are treated as an iterable (thus iterating through the keys), but you're right. Updated that in the last commit

File: python/ray/tune/tests/test_progress_reporter.py (hunk 3 above)
Comment (krfricke) [reply python/ray/tune/tests/test_progress_reporter.py:262]: They come up in examples such as #9307. I also tested this with a manual config/constant values, and it still works. The flattened list lookup comes from `flatten_resolved_vars()`.
I amended the PR to allow lookup of other Sequences, too, and fixed a small bug where a lookup of `a/b/c` would return `lookup["a"]["b"]` if `lookup["a"]["b"]["c"]` did not exist. In this case, `default` should be returned.

**Comment by krfricke** [comment]
As an additional note, this PR introduces some flexibility in the reporting. E.g. if your evaluated config is
```
config = {
    "a": {
        "b": (3, 4)
    }
}
```

Then a lookup of `"a/b/0"` would return `3`, but a lookup of `"a/b"` would still be possible and return `(3, 4)`. This allows for some flexibility when selecting reporting metrics.

**Pull request merged**